reviews/source_control/models.py

```diff
@@ -22,3 +24,50 @@ class PullRequest:
     approved: str
     approved_by_others: bool
     labels: List[Label] = field(default_factory=list)
+
+    def render_approved(self) -> str:
+        """Renders the approved status as a colourised string"""
+        if self.approved == "APPROVED":
+            return "[green]Approved[/]"
+        if self.approved == "CHANGES_REQUESTED":
+            return "[red]Changes Requested[/]"
+        return ""
+
+    def render_approved_by_others(self) -> str:
+        """Renders approved_by_others flag as a colourised string"""
+        return "[green]Ready[/]" if self.approved_by_others else ""
+
+    def render_labels(self, colour_map: Dict[str, str]) -> str:
+        """Renders the labels as a joined colourised string"""
+        return ", ".join(
+            [
+                f"{colour_map.get(label.name.lower(), '[white]')}{label.name}"
+                for label in self.labels
+            ]
+        )
+
+    def render_title(self, org: str, repository: str) -> str:
```

Comment:
NAB: It would be fantastic to add unit tests to cover these functions. As they don't already exist for the existing code please consider it non-blocking but an example of how to add tests for these functions may look like:

```python
from datetime import datetime

from freezegun import freeze_time
import pytest

from reviews.datasource import PullRequest, Label


@pytest.fixture
def organisation() -> str:
    return "my-org"

@pytest.fixture
def repository() -> str:
    return "my-repo"


@pytest.fixture
@freeze_time("2020-01-01T00:00:00+00:00")
def pull_request():
    return PullRequest(
        number=1,
        draft=True,
        title="[1] Initial Commit",
        created_at=datetime.now(),
        updated_at=datetime.now(),
        approved=None,
        approved_by_others=False,
        labels=[Label(name="Python")],
    )


def test_pull_request_renders_title(organisation, repository, manager):
    title = pull_request.render_title(org=organisation, repository=repository) 

    assert title == "[Draft] [1] Initial Commit"
```